File: ios/Classes/NativeIdPlugin.m
#import "NativeIdPlugin.h"
#if __has_include(<native_id/native_id-Swift.h>)
#import <native_id/native_id-Swift.h>
#else
#import "native_id-Swift.h"
#endif

@implementation NativeIdPlugin

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [SwiftNativeIdPlugin registerWithRegistrar:registrar];
}

@end

File: ios/Classes/NativeIdPlugin.swift
import Flutter
import UIKit

public class NativeIdPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "native_id", binaryMessenger: registrar.messenger())
    let instance = NativeIdPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getId":
      let identifierForVendor = UIDevice.current.identifierForVendor?.uuidString
      result(identifierForVendor)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
